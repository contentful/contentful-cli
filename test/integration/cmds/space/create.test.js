import nixt from 'nixt'
import { join } from 'path'
import {
  initConfig,
  deleteSpaces,
  extractSpaceId
} from '../../util'

const bin = join(__dirname, './../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true }).cwd(bin).base('./contentful.js ').clone()
}

var spacesToDelete = []

beforeAll(() => {
  return initConfig()
})

afterAll(() => {
  return deleteSpaces(spacesToDelete)
}, 10000)

test('should exit 1 when no args', done => {
  app()
    .run('space create')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(() => {
      done()
    })
})

test('should print help message', done => {
  app()
    .run('space create --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should create space', done => {
  app()
    .run(`space create --name cli_test_org_space --organization-id ${process.env.CLI_E2E_ORG_ID}`)
    .expect((result) => {
      const resultText = result.stdout.trim()
      var spaceId = extractSpaceId(resultText)
      spacesToDelete.push(spaceId)
      const regex = /Successfully created space .*/ // name doesn't matter because response is replayed
      expect(result.stdout.trim()).toMatch(regex)
    })
    .code(0)
    .end(done)
})

test('should create space using shortcuts args', done => {
  app()
    .run(`space create -n cli_test_org_space_sh --org ${process.env.CLI_E2E_ORG_ID}`)
    .expect((result) => {
      const resultText = result.stdout.trim()
      var spaceId = extractSpaceId(resultText)
      spacesToDelete.push(spaceId)
      const regex = /Successfully created space .*/ // name doesn't matter because response is replayed
      expect(result.stdout.trim()).toMatch(regex)
    })
    .code(0)
    .end(done)
})
