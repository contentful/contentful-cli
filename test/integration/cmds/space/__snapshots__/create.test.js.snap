// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should exit 1 when no args: help data is incorrect 1`] = `
"Usage: contentful space create --name 'Your Space Name'

Options:
  -h, --help                Show help                                  [boolean]
  --name, -n                Name of the space to create               [required]
  --management-token, --mt  Contentful management API token             [string]
  --organization-id, --org  Organization owning the new space
  --yes, -y                 Confirm space creation without prompt, be aware this
                            may result in extra monthly charges depend on your
                            subscription
  --default-locale, -l      The default locale of the new space         [string]
  --use, -u                 Use the created space as default space when the
                            --space-id is skipped.                     [boolean]

Copyright 2019 Contentful
Missing required argument: name"
`;

exports[`should print help message: help data is incorrect 1`] = `
"Usage: contentful space create --name 'Your Space Name'

Options:
  -h, --help                Show help                                  [boolean]
  --name, -n                Name of the space to create               [required]
  --management-token, --mt  Contentful management API token             [string]
  --organization-id, --org  Organization owning the new space
  --yes, -y                 Confirm space creation without prompt, be aware this
                            may result in extra monthly charges depend on your
                            subscription
  --default-locale, -l      The default locale of the new space         [string]
  --use, -u                 Use the created space as default space when the
                            --space-id is skipped.                     [boolean]

Copyright 2019 Contentful"
`;
